### PR TITLE
968: Added a new context dependent Action for the webapi.xml file generation

### PR DESCRIPTION
**Pull request by doninAtwix**
**Description** (*)
968: Added a new context dependent Action for the webapi.xml file generation

**Fixed Issues (if relevant)**

1. Fixes magento/magento2-phpstorm-plugin#968

**Questions or comments**

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/context/xml/NewWebapiXmlAction.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.magento.idea.magento2plugin.util.magento.GetMagentoModuleUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class NewWebapiXmlAction extends AbstractContextAction {
+
+    public static final String ACTION_NAME = "Magento 2 WEBAPI File";
+    public static final String ACTION_DESCRIPTION = "Create a new Magento 2 webapi.xml file";
+
+    /**
+     * New routes.xml file generation action constructor.
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/context/xml/NewWebapiXmlAction.java:24]: Please, change the annotation to the appropriate one.

File: src/com/magento/idea/magento2plugin/actions/context/xml/NewWebapiXmlAction.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.magento.idea.magento2plugin.actions.context.AbstractContextAction;
+import com.magento.idea.magento2plugin.magento.files.ModuleWebApiXmlFile;
+import com.magento.idea.magento2plugin.magento.packages.ComponentType;
+import com.magento.idea.magento2plugin.magento.packages.Package;
+import com.magento.idea.magento2plugin.util.magento.GetMagentoModuleUtil;
+import org.jetbrains.annotations.NotNull;
+
+public class NewWebapiXmlAction extends AbstractContextAction {
+
+    public static final String ACTION_NAME = "Magento 2 WEBAPI File";
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/context/xml/NewWebapiXmlAction.java:20]:
```suggestion
    public static final String ACTION_NAME = "Magento 2 WEB API File";
```

**Review by bohdan-harniuk — CHANGES_REQUESTED** [review]
Hello, @doninAtwix!

Please, check my suggestions + update your branch from the mainline.

File: src/com/magento/idea/magento2plugin/actions/context/xml/NewWebapiXmlAction.java (hunk 1 above)
Comment (doninAtwix) [reply src/com/magento/idea/magento2plugin/actions/context/xml/NewWebapiXmlAction.java]: Hi @bohdan-harniuk .
Thanks for your code review. I fixed. Please check.

**Review by bohdan-harniuk — APPROVED** [review]
Hello, @doninAtwix!

Thank you for your work here!

**Review by eduard13 — APPROVED** [review]

**Pull request merged**